File: services/order-service/src/schemas/ingredients.py
from datetime import datetime
from typing import Optional
from pydantic import BaseModel


class IngredientBase(BaseModel):
    name: str
    quantity_on_hand: int
    restaurant_id: int
    unit: str
    reorder_point: int


class IngredientCreate(IngredientBase):
    pass


class IngredientUpdate(BaseModel):
    name: Optional[str] = None
    quantity_on_hand: Optional[int] = None
    restaurant_id: Optional[int] = None
    unit: Optional[str] = None
    reorder_point: Optional[int] = None


class Ingredient(IngredientBase):
    id: int

    class ConfigDict:
        from_attributes = True

File: apps/pos-system/src/lib/api/endpoints/ingredients.ts
import { apiClient } from '../client.js';

export interface Ingredient {
    id: number;
    name: string;
    quantity_on_hand: number;
    apriltag_id: number | null;
    restaurant_id: number;
    unit: string;
    reorder_point: number;
}

export interface IngredientUpdate {
    name?: string;
    quantity_on_hand?: number;
    apriltag_id?: number | null;
}

export const ingredientsApi = {
    async getAll(): Promise<Ingredient[]> {
        return apiClient.request<Ingredient[]>('ingredients');
    },

    async update(id: number, data: IngredientUpdate): Promise<Ingredient> {
        return apiClient.request<Ingredient>(`ingredients/${id}`, {
            method: 'PUT',
            body: JSON.stringify(data)
        });
    }
};
